File: src/Q57.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    vector<int> ret;
    for (auto l = nums.begin(), r = nums.end() - 1; l < r;) {
      if (*l + *r == target) {
        ret.push_back(*l);
        ret.push_back(*r);
        break;
      } else if (*l + *r < target) {
        l++;
      } else {
        r--;
      }
    }
    return ret;
  }
};

File: src/Q57.rs
struct Solution;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::cmp::Ordering;
        let (mut l, mut r) = (0, nums.len() - 1);
        while l < r {
            match (nums[l] + nums[r]).cmp(&target) {
                Ordering::Equal => return vec![nums[l], nums[r]],
                Ordering::Less => l += 1,
                Ordering::Greater => r -= 1,
            }
        }
        Vec::new()
    }
}
